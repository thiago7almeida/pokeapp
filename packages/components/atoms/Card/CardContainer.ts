import styled from 'styled-components/native';
import {responsive, spacing} from '@components';

export const CardContainer = styled.View<{backgroundColor: string}>`
  background-color: ${({backgroundColor}) => backgroundColor};
  border-radius: ${spacing.s16}px;
  padding: ${spacing.s8}px;
  flex-grow: 1;
  height: ${responsive.size(160, 'vertical')}px;
`;
