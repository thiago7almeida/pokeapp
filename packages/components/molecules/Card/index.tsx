import React from 'react';

import {CardContainer, Typography, FloatImage} from '@components';

export const Card: React.FC<{
  title: string;
  backgroundColor: string;
  uri: string;
}> = ({title, backgroundColor, uri}) => {
  return (
    <CardContainer backgroundColor={backgroundColor}>
      <Typography.Label>{title}</Typography.Label>
      <FloatImage source={{uri}} />
    </CardContainer>
  );
};
