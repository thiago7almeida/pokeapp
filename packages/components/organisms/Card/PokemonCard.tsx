import React, {useMemo} from 'react';

import {colors, Card} from '@components';
import {Pressable} from 'react-native';

export const PokemonCard: React.FC<{
  pokemon: {name: string; type: string; image: string};
  onPress: () => void;
}> = ({pokemon, onPress}) => {
  const backgroundColor = useMemo(() => {
    return colors[pokemon.type] || colors.bug;
  }, [pokemon]);

  return (
    <Pressable onPress={onPress}>
      <Card
        backgroundColor={backgroundColor}
        title={pokemon.name}
        uri={pokemon.image}
      />
    </Pressable>
  );
};
