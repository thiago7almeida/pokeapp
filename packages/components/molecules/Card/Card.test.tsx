import React from 'react';
import {render} from '@testing-library/react-native';
import {Card} from './Card';
import '@testing-library/jest-native/extend-expect';

describe('Card', () => {
  const title = 'Card Title';
  const backgroundColor = 'red';
  const uri = 'https://example.com/image.png';

  it('should render correctly', () => {
    const {toJSON} = render(
      <Card title={title} backgroundColor={backgroundColor} uri={uri} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the title', () => {
    const {getByText} = render(
      <Card title={title} backgroundColor={backgroundColor} uri={uri} />,
    );
    const titleElement = getByText(title);

    expect(titleElement).toBeDefined();
  });

  it('should set the background color', () => {
    const {getByTestId} = render(
      <Card title={title} backgroundColor={backgroundColor} uri={uri} />,
    );

    expect(getByTestId('card-container')).toHaveStyle({backgroundColor});
  });

  it('should render the image', () => {
    const {getByTestId} = render(
      <Card title={title} backgroundColor={backgroundColor} uri={uri} />,
    );
    const image = getByTestId('card-image');

    expect(image.props.source.uri).toBe(uri);
  });
});
