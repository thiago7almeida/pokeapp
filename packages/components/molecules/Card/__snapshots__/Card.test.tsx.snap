// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<View
  backgroundColor="red"
  style={
    [
      {
        "backgroundColor": "red",
        "borderRadius": 29,
        "flexGrow": 1,
        "height": 267,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      },
    ]
  }
  testID="card-container"
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 29,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Card Title
  </Text>
  <Image
    source={
      {
        "uri": "https://example.com/image.png",
      }
    }
    style={
      [
        {
          "bottom": -15,
          "height": 234,
          "position": "absolute",
          "right": -15,
          "width": 234,
        },
      ]
    }
    testID="card-image"
  />
</View>
`;
